Wrangler2 release (#3909)
cloudflare/cloudflare-docs: content/workers/wrangler/migration/deprecations.md

---
pcx-content-type: how-to
title: Deprecations
weight: 2
---

# Deprecations

> This document describes the difference between `wrangler` versions 1.x and 2.0.0, specifically deprecations and breaking changes.

Reviewer: ```suggestion
This document describes the difference between `wrangler` versions 1 and 2, specifically deprecations and breaking changes.
```

<!-- - **`legacy_env`**: _boolean_, optional. default: `true`

  The `legacy_env` field toggles how environments are handled by `wrangler`.

  - When `legacy_env` is `true`, it uses the legacy-style environments, where each environment is treated as a separate Worker in the dashboard, and environment names are appended to the `name` when published.
  - When `legacy_env` is `false`, it uses the newer service environments, where scripts for a given Worker are grouped under the same script name in the Cloudflare Workers dashboard, and environments are subdomains for a given published script (when `workers_dev = true`).
    Read more at (ref:)[] -->

<!-- - **`services`**: TODO

- **`node-compat`**: TODO

- **`public`**: TODO -->

### Non-mandatory fields

> A few configuration fields which were previously required, are now optional in particular situations; they can either be inferred, or added as an optimisation. No fields are required anymore when starting with `wrangler` version 2.0.0, and you can gradually add configuration as the need arises.

Reviewer: ```suggestion
A few configuration fields which were previously required, are now optional in particular situations; they can either be inferred, or added as an optimization. No fields are required anymore when starting with `wrangler` version 2, and you can gradually add configuration as the need arises.
```

- **`name`**: _string_

  The `name` configuration field is now not required for `wrangler dev`, or any of the `wrangler kv:*` commands. Further, it can also be passed as a command line argument as `--name <name>`. It is still required for `wrangler publish`.

- **`account_id`**: _string_

>   The `account_id` field is not required for any of the commands. Any relevant commands will check if you are logged in, and if not, will prompt you to log in. Once logged in it will use your account ID, and will not prompt you again until your login session expires. If you have multiple account IDs, you will be presented with a list of accounts to choose from.

Reviewer: ```suggestion
  The `account_id` field is not required for any of the commands. Any relevant commands will check if you are logged in, and if not, will prompt you to log in. Once logged in, it will use your account ID and will not prompt you again until your login session expires. If you have multiple account IDs, you will be presented with a list of accounts to choose from.
```

>   You can still configure `account_id` in your `wrangler.toml` file, or as an environment variable `CLOUDFLARE_ACCOUNT_ID`; this will make startup faster, and bypass the list of choices if you have multiple IDs. The `CLOUDFLARE_API_TOKEN` environment variable is also useful for situations where it is not possible to login interactively - see [Running in CI/CD](/workers/wrangler/ci-cd).

Reviewer: ```suggestion
  You can still configure `account_id` in your `wrangler.toml` file, or as an environment variable `CLOUDFLARE_ACCOUNT_ID`; this will make startup faster, and bypass the list of choices if you have multiple IDs. The `CLOUDFLARE_API_TOKEN` environment variable is also useful for situations where it is not possible to login interactively - refer to [Running in CI/CD](/workers/wrangler/ci-cd).
```

- **`workers_dev`** _boolean_, default: `true` when no routes are present

>   The `workers_dev` field is used to indicate that the Worker should be published to a `*.workers.dev` subdomain. For example, for a Worker named my-worker and a previously configured `*.workers.dev` subdomain `username`, the Worker will get published to `my-worker.username.workers.dev.com`. This field is not mandatory, and defaults to `true` when `route` or `routes` are not configured. When routes are present, it defaults to `false`. If you want to neither publish it to a `*.workers.dev` subdomain nor any routes, set `workers_dev` to `false`. This useful when you are publishing a Worker as a standalone service that can only be accessed from another Worker with (`services`)[TODO:link/to/services/docs] bindings.

Reviewer: ```suggestion
  The `workers_dev` field is used to indicate that the Worker should be published to a `*.workers.dev` subdomain. For example, for a Worker named `my-worker` and a previously configured `*.workers.dev` subdomain `username`, the Worker will get published to `my-worker.username.workers.dev.com`. This field is not mandatory, and defaults to `true` when `route` or `routes` are not configured. When routes are present, it defaults to `false`. If you want to neither publish it to a `*.workers.dev` subdomain nor any routes, set `workers_dev` to `false`. This useful when you are publishing a Worker as a standalone service that can only be accessed from another Worker with (`services`)[TODO:link/to/services/docs] bindings.
```

- **`zone_id`**: _string_, deprecated

  The `zone_id` field is deprecated and will be removed in a future release. It is now inferred from `route`/`routes`, and optionally from `dev.host` when using `wrangler dev`. This also makes it simpler to deploy a single Worker to multiple domains.

- **`build.upload`**: _object_, deprecated

  The `build.upload` field is deprecated and will be removed in a future release. Its usage results in a warning with suggestions on rewriting the configuration file to remove the warnings.

  - `build.upload.main`/`build.upload.dir` are replaced by the `main` fields and are applicable to both service worker format and modules format Workers.
  - `build.upload.rules` is replaced by the `rules` field and is applicable to both service worker format and modules format Workers.
  - `build.upload.format` is no longer specified and is automatically inferred by `wrangler`.

### Deprecated fields (breaking)

A few configuration fields are deprecated and will not work as expected anymore. It is recommended to read the error messages and follow the instructions to migrate to the new configuration.

- **`site.entry-point`**: `string`, deprecated

>   The `site.entry-point` configuration was used to specify an entry point for Workers with a `[site]` configuration. This has been replaced by the top level `main` field.

Reviewer: ```suggestion
  The `site.entry-point` configuration was used to specify an entry point for Workers with a `[site]` configuration. This has been replaced by the top-level `main` field.
```

- **`type`**: `rust` | `javascript` | `webpack`, deprecated

>   The `type` configuration was used to specify the type of Worker; it has since been made redundant and is now inferred from usage. If you were using `type = "webpack"` (and the optional `webpack_config` field), you should read the [TODO: webpack migration guide](#todo/webpack-migration.md) to modify your project and use a custom build instead.

Reviewer: ```suggestion
  The `type` configuration was used to specify the type of Worker. It has since been made redundant and is now inferred from usage. If you were using `type = "webpack"` (and the optional `webpack_config` field), you should read the [TODO: webpack migration guide](#todo/webpack-migration.md) to modify your project and use a custom build instead.
```

## Deprecated commands

> The following commands are deprecated in Wrangler as of v2.0.0.

Reviewer: ```suggestion
The following commands are deprecated in Wrangler as of Wrangler 2.
```

Create the `workers.dev` subdomain on your Workers dashboard.

### generate:

The `wrangler generate` command is no longer available to initialize a Worker from a template repository.

Initialize a basic Worker project via `wrangler init`, which provides a good starting point for most projects.

Alternatively, try cloning the template repository and initializing it manually.

### route:

The `wrangler route` command is no longer available to configure a route for a Worker.

Routes are specified in the `wrangler.toml` configuration file.

## Other deprecated behaviour

- `wrangler` will no longer use `index.js` in the directory where `wrangler dev` is called as the entry point to a Worker. Use the `main` configuration field, or explicitly pass it as a command line argument, for example: `wrangler dev index.js`.

- `wrangler` will no longer assume that bare specifiers are file names if they are not represented as a path. For example, in a folder like so:

  ```
  project
  ├── index.js
  └── some-dependency.js
  ```

  where the content of `index.js` is:

  ```js
  import SomeDependency from "some-dependency.js";

  addEventListener("fetch", (event) => {
    // ...
  });
  ```

>   `wrangler` 1.x would resolve `import SomeDependency from "some-dependency.js";` to the file `some-dependency.js`. This will also work in wrangler 2.0, but will also log a deprecation warning; in the future, this will break with an error. Instead, you should rewrite the import to specifiy that it is a relative path, like so:

Reviewer: ```suggestion
  `wrangler` 1 would resolve `import SomeDependency from "some-dependency.js";` to the file `some-dependency.js`. This will also work in `wrangler` 2, but will also log a deprecation warning; in the future, this will break with an error. Instead, you should rewrite the import to specifiy that it is a relative path, like so:
```
